library/java/src/io/envoyproxy/envoymobile/engine/EnvoyEngine.java

```diff
@@ -72,5 +73,15 @@ int runWithConfig(EnvoyConfiguration envoyConfiguration, String logLevel,
    */
   int recordGaugeSub(String elements, int amount);
 
+  /**
+   * Add another recorded amount to the histogram with the given string of elements.
+   *
+   * @param elements Elements of the histogram stat.
+   * @param value Amount to record as a new value for the histogram distribution.
+   * @param unitMeasure the unit of measurement (e.g. milliseconds, bytes, etc.)
+   * @return A status indicating if the action was successful.
+   */
+  int recordHistogramValue(String elements, int value, HistogramUnit unitMeasure);
```

Comment:
it probably worth documenting (as a comment in this pr) the reason of pushing `HistogramUnit` enum to this layer (instead of keeping in below jni)

Reply:
I updated to hide the enum at the JNI layer. I think it's cleaner to have 2 separate implementations here.